shell/common/api/electron_api_native_image_mac.mm

```diff
@@ -39,20 +39,37 @@
   return def;
 }
 
+CGSize getSizefromPath(const base::FilePath& path, const gfx::Size& max_size) {
+  base::ScopedCFTypeRef<CFURLRef> url = base::mac::FilePathToCFURL(path);
+  base::ScopedCFTypeRef<CGImageSourceRef> source(
+      CGImageSourceCreateWithURL(url, nullptr));
```

Comment:
I think `CreateThumbnailFromPath` API means to get thumbnail from any type of file, while `getSizefromPath` can only get size of image files?

Reply:
Can you explain more what you mean? `getSizefromPath` returns the default size of the image. The `maxSize` param is intended to ensure an image is _no larger_ than the specified size, but instead set it to that size _regardless_ of the size of the initial image. `getSizefromPath` was added to retrieve the initial image size, so the returned thumbnail with `CreateThumbnailFromPath` would be _either_ maxSize or the default, size, whichever is smaller.

Reply:
But `getSizefromPath` can only get sizes of images files, so if you call `createThumbnailFromPath('text.word')` it can not return the size of `text.word`'s thumbnail's size because `CGImageSourceCreateWithURL('text.word')` will fail.

Reply:
Ah, i see what you're saying. I suppose I could only trigger this for images?

Reply:
Only triggering for images probably won't be very useful, I think people are mostly using `createThumbnailFromPath` on non-image files.

Reply:
sure, but as-is the `maxSize` parameter does not work as described for images - the behavior is incorrect, and I think it should be corrected to match what the documentation says it should conform to

Reply:
Fixing the issue is good, but partially fixing it is probably not so. With current change we have to add something like this in docs:

> On macOS the maxSize is respected if passed file is an macOS API image file (such as JPG/PNG/PDF but not SVG/WEBP), otherwise the the thumbnail will be always resized to maxSize.

If users are only using `createThumbnailFromPath` to generate thumbnails for certain kinds of image files, then they would be happy to see this one case to be fixed. But the `createThumbnailFromPath` API is mostly used to generate preview for any kind of type file, so for most users they now have to branch their code because of the behavior change:

```js
if (path.is('png', 'jpg')) {
  // new code that pass maxSize
} else {
  // old code that pass size
}
```

Which feels more like a burden than a feature.